yin_yang/position.py

```diff
@@ -1,13 +1,55 @@
 import logging
-from time import sleep
 
 import requests
-from PySide6.QtCore import QObject
-from PySide6.QtPositioning import QGeoPositionInfoSource, QGeoCoordinate, QGeoPositionInfo
+from PySide6.QtCore import QObject, Slot
+from PySide6.QtPositioning import (
+    QGeoPositionInfoSource,
+    QGeoCoordinate,
+    QGeoPositionInfo,
+)
 
 logger = logging.getLogger(__name__)
 
 
+class QTPositionReceiver(QObject):
+    """Small handler for QT positioning service."""
+
+    def __init__(self):
+        super().__init__()
+
+        # Create the position service and hook it to us
+        self._positionSource = QGeoPositionInfoSource.createSource(
+            "geoclue2", {"desktopId": "Yin-Yang"}, self
+        )
+        self._positionSource.positionUpdated.connect(self.handlePosition)
+
+        # Start the position service.
+        # This will only work after the app calls `exec()`.
+        self._positionSource.startUpdates()
+
+        # Get the initial last position.
+        self._lastPosition = self._positionSource.lastKnownPosition()
+
+    @Slot(QGeoPositionInfo)
+    def handlePosition(self, position: QGeoPositionInfo):
+        """Track the position provided by the service."""
+        self._lastPosition = position
+
+        # TODO: Here trigger a refresh.
```

Comment:
Are you still figuring that out? If so, please mark this PR as a draft

Reply:
Whoops, just read the comment in the PR.

The location in the config file is never changed automatically, it's always the user-defined value. If the location is automatically determined, we use the functions directly instead of the value in the file. If the detection fails, we use the user-defined value as a fallback.

Reply:
I believe a good thing would be to update the value in the config too.
At least, that is what I would expect if I select the "auto" option...
However, the choice is yours and if you say "no", then the code is complete and I just need to remove the comment :wink: